### PR TITLE
Uar 1178 delete stmt validation flag

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-1178

### Change description
Removed all instances of the FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION feature flag and replaced with the value or code code that is covered in the "true" scenario.

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/check.your.answers.ts [hunk 1]
```diff
@@ -153,11 +152,13 @@ const getBackLinkUrl = (isNoChangeJourney: boolean, hasAnyBosWithTrusteeNocs: bo
     return UPDATE_NO_CHANGE_REGISTRABLE_BENEFICIAL_OWNER_URL;
   } else {
     let backLinkUrl: string;
-    backLinkUrl = (isActiveFeature(FEATURE_FLAG_ENABLE_UPDATE_TRUSTS) && hasAnyBosWithTrusteeNocs) ? UPDATE_TRUSTS_ASSOCIATED_WITH_THE_OVERSEAS_ENTITY_URL : UPDATE_BENEFICIAL_OWNER_TYPE_URL;
+    backLinkUrl = (isActiveFeature(FEATURE_FLAG_ENABLE_UPDATE_TRUSTS) && hasAnyBosWithTrusteeNocs) ?
+      UPDATE_TRUSTS_ASSOCIATED_WITH_THE_OVERSEAS_ENTITY_URL :
+      UPDATE_BENEFICIAL_OWNER_TYPE_URL;
```
Comment (markpit) [inline src/utils/check.your.answers.ts:157]: line 155 - 157 is causing a code smell, it's my fault as I thought this section was still needed and advised it to be re-added. But looking at the flow of code, the backLinkUrl gets overwritten by the if(isRemove) / else blocks.
So lines 155-157 can be removed, that should fix the code smell

File: test/controllers/update/no.change.registrable.beneficial.owner.controller.spec.ts [hunk 2]
```diff
@@ -131,7 +131,7 @@ describe("No change registrable beneficial owner", () => {
         .send({ [RegistrableBeneficialOwnerKey]: "1" });
 
       expect(resp.status).toEqual(302);
-      expect(resp.header.location).toEqual(UPDATE_REVIEW_STATEMENT_URL);
+      expect(resp.header.location).toEqual(UPDATE_STATEMENT_VALIDATION_ERRORS_URL);
```
Comment (markpit) [inline test/controllers/update/no.change.registrable.beneficial.owner.controller.spec.ts:134]: the test description needs updating for these 2 tests, it still says should redirect to UPDATE_REVIEW_STATEMENT_URL

File: src/utils/check.your.answers.ts (hunk 1 above)
Comment (mattch1) [reply src/utils/check.your.answers.ts]: Removed this.

File: test/controllers/update/no.change.registrable.beneficial.owner.controller.spec.ts (hunk 2 above)
Comment (mattch1) [reply test/controllers/update/no.change.registrable.beneficial.owner.controller.spec.ts:133]: Changed description

File: test/controllers/update/overseas.entity.update.details.controller.spec.ts [hunk 3]
```diff
@@ -238,7 +238,7 @@ describe("OVERSEAS ENTITY UPDATE DETAILS controller", () => {
         .send(UPDATE_ENTITY_BODY_OBJECT_MOCK_WITH_ADDRESS);
 
       expect(resp.status).toEqual(302);
-      expect(resp.text).toContain(config.BENEFICIAL_OWNER_STATEMENTS_PAGE);
+      expect(resp.text).toContain(config.UPDATE_BENEFICIAL_OWNER_BO_MO_REVIEW_URL);
```
[the remark this answers is not in the document]
Comment (mattch1) [reply test/controllers/update/overseas.entity.update.details.controller.spec.ts:241]: Changed description

**Review by markpit — APPROVED** [review]

**Pull request merged**